### PR TITLE
Delete free extensions transient on WCA update

### PR DESCRIPTION
**Pull request by joshuatf**
I think this is an edge case, but just in case users have already cached values for free extensions, this deletes that transient when WooCommerce Admin is updated.

Since there are breaking changes between the v1.0 and v2.0 REST API, this just confirms the data is fetched again using the new API.

### Detailed test instructions:

1. Visit the marketing task or business details step to set the transient
2. Modify the value of the transient at `woocommerce_admin_remote_free_extensions_specs` to something you can recognize.
3. Bump the WooCommerce Admin version.
4. Note that the transient is deleted and a new one is stored when visiting the task.

### CODE REVIEW

File: src/Features/RemoteFreeExtensions/Init.php [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Init {
 	 */
 	public function __construct() {
 		add_action( 'change_locale', array( __CLASS__, 'delete_specs_transient' ) );
+		add_action( 'woocommerce_admin_updated', array( __CLASS__, 'delete_specs_transient' ) );
```
Comment (louwie17) [inline src/Features/RemoteFreeExtensions/Init.php:24]: @joshuatf I believe this constructor is never called, as all the other functions are static, so it doesn't get instantiated anywhere? Or at-least my break points never hit this, or an error log I added.

**Review by louwie17 — CHANGES_REQUESTED** [review]
As mentioned below, the constructor for `Init.php` is never called ( I think, as this could also be my lack of PHP knowledge )

File: src/Features/RemoteFreeExtensions/Init.php (hunk 1 above)
Comment (joshuatf) [reply src/Features/RemoteFreeExtensions/Init.php:24]: @louwie17 Good call.  I think the feature name was incorrect, causing this constructor to never get called.  

Updated this; note that `npm run build:feature-config` will need to be run to rebuild the feature config file.

Comment (joshuatf) [reply src/Features/RemoteFreeExtensions/Init.php:24]: Quick note: I also didn't update this feature name in the business details step given it's being removed from there in https://github.com/woocommerce/woocommerce-admin/pull/7452.

Comment (louwie17) [reply src/Features/RemoteFreeExtensions/Init.php:24]: Thanks, that indeed fixed it :)

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 It tested well after the feature name update, nice work!!

**Pull request merged**